src/ListingPage.php

```diff
@@ -192,4 +188,53 @@ public function view(): void
         do_action('enqueue_google_tag_manager_script', $this->context);
         Timber::render($this->templates, $this->context);
     }
+
+    public static function get_sticky_posts(): array
+    {
+        $sticky_posts = get_option('sticky_posts');
+
+        if (empty($sticky_posts)) {
+            return [];
+        }
+
+        $sticky_posts = new \WP_Query([
+            'post__in' => $sticky_posts,
+            'posts_per_page' => self::$MAX_STICKY_POSTS,
+            'post_type' => 'post',
+            'post_status' => 'publish',
+            'fields' => 'ids',
+            'orderby' => 'date',
+            'order' => 'DESC',
+        ]);
+
+        if (count($sticky_posts->posts) < self::$MAX_STICKY_POSTS) {
+            return [];
+        }
+
+        return array_slice($sticky_posts->posts, 0, self::$MAX_STICKY_POSTS);
+    }
+
+    /**
+     * Move the categories filter to the ListingPage class.
+     */
+    public static function get_selected_categories(): array
+    {
+        $category_slug = isset($_GET['category']) ? $_GET['category'] : '';
+        $category = get_category_by_slug($category_slug);
+        return $category ? [$category->term_id] : [];
+    }
+
+    /**
+     * Move the post types filter to the ListingPage class.
+     */
+    public static function get_selected_post_types(): array
```

Comment:
I moved out this functionality from `functions.php` into the proper file and reduce the complexity of `functions.php`.

Using a `static` function to let us use it whithout to intiliaze the `ListingPage Class`.